Public/Sdk/Public/Transformers/Transformer.Execute.dsc

```diff
@@ -114,6 +114,9 @@ namespace Transformer {
         /** Set outputs to remain writable */
         keepOutputsWritable?: boolean;
 
+        /** Whether this process requires admin privilege */
+        requiresAdmin?: boolean;

```

Comment:
Can we make this an Enum for the future:

`adminElevation: "none" | "required" | "desired"`

At the moment we will run all child processes as the elevation level of the engine. We should go to a mode that even if the engine is elevated all child processes will run as not elevated to protect developers.
the future "desired" can decide to run the pip as admin when he engine is run as admin and unelevated if the engine is unelevated
"required" would mean that if the engine is unelevated this pip will fail unless your new feature is enabled on the commandline to remote the work.

Reply:
Mike and I discuss the run unelevated case, and we doubt that the scenario will appear. I'll start simple with boolean. This simplicity allows me to use Process.Options (no changes to Process.cs and ProcessBuilder.cs). 

Also the enforcement of the required-admin flag will be done using the runexternal feature. If internal or external tool, then it runs with the elevation level of the engine. If VM, then it runs with VM admin privilege.

For dev build, the default will be internal (we don't expect people to always have VM ready). In this case, you can think that we run the required-admin pip best effort as admin. In CB, the default will be VM.

For tests that require admin, if we run in VM, then the tests should fail if they are not run under admin in VM. This last feature still need to be done.

---
In reply to: [276810187](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276810187) [](ancestors = 276810187)

Reply:
Interally you can keep a boolean, on the flags no problem!
I just want our API to be future-proof. Also just in case we get multiple ways to do admin, maybe no the mac it will be different and one way will work or not.

Let's just feature-proof it.
I'm happy to do the work for you if that is the problem, just let me know and I'll pusht he change to your branch.

"Waiting" for a response on that for now.



---
In reply to: [277062597](https://github.com/Microsoft/BuildXL/pull/159#discussion_r277062597) [](ancestors = 277062597,276810187)